docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -31,21 +31,102 @@
 
 .. _build-arrow:
 
-*********************************
-Building Arrow's libraries 🏋🏿‍♀️
-*********************************
+************************************
+Building the Arrow's libraries 🏋🏿‍♀️
+************************************
 
+The Arrow project contains good number of libraries that enable
+work in many languages. Most libraries (C++, C#, Go, Java,
+JavaScript, Julia, and Rust) already contain distinct implementations
+of Arrow. 
 
+It is different for C (Glib), MATLAB, Python, R, and Ruby as they
+are built on top of the C++ library. In this section we will be
+dealing with this second part of the libraries plus some of C++.
 
-Building C++
-============
+In this case, if you decide to contribute to Arrow you will meet
+the topic of compiling the source code and use of CMake. You may
+have some experience with it or not. If not, it is good to read
+through this part so you understand what is happening in the process
+of building Arrow better.
```

Comment:
```suggestion
If you decide to contribute to Arrow you might
need to compile the C++ source code.  This is done using a tool called CMake, which you may or may not have experience with.  If not, this section of the guide will help you better understand CMake and the process
of building Arrow's C++ code.
```

Not all contributors will need to compile the C++; in the R developer docs we have instructions for using the nightly builds as the C++ source when people are only making changes to the R code without needing to build C++ from source.